network/tailscale/pkg.yaml

```diff
@@ -0,0 +1,80 @@
+name: tailscale
+variant: scratch
+shell: /toolchain/bin/bash
+dependencies:
+  - stage: base
+steps:
+  - env:
+      GOPATH: /go
+  - sources:
+      - url: https://github.com/tailscale/tailscale/archive/refs/tags/v{{ .TAILSCALE_VERSION }}.tar.gz
+        destination: tailscale.tar.gz
+        sha256: 6964176889943e0e0b25d8d69e14226cfb1c1a9944a257b24cb2dd212f797141
+        sha512: a64cad7685ba1dd438e16c8fa6ed1c29e5d840ffe17b2417342e43a2da5c1679826eb3a29b33e40ac075171a59605cdc857e1e09e60cf5140b22d363bb31ff30
+      - url: https://skarnet.org/software/s6/s6-{{ .S6_VERSION }}.tar.gz
```

Comment:
i would avoid adding all these dependencies and write  a simple wrapper like here: https://github.com/siderolabs/extensions/tree/main/storage/iscsi-tools/iscsid-wrapper

It just need to loop over files in a directory and set env vars from it

Reply:
Fair, it did feel getting excessive :). I don't know much go, and have used the s6 tools a lot before, so reached for what I know.

I'm also hoping any wrapper should also be short-lived, as if extensions supported envFiles, then it wouldn't be as necessary. I raised an issue for that. https://github.com/siderolabs/talos/issues/7316.

It may take me awhile to rewrite into a go wrapper, as I have no experience with go at the moment, and my system extension testing involves deploying to a RPI, so it's not the best turnaround, but I will give it a try

Reply:
@btrepp I've added the envfile support and cleaned up the PR. Sorry for the long wait.